kernel/src/hil/time.rs

```diff
@@ -578,3 +648,133 @@ impl PartialEq for Ticks64 {
 }
 
 impl Eq for Ticks64 {}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    struct Test1MHz64();
+    impl Time for Test1MHz64 {
+        type Frequency = Freq1MHz;
+        type Ticks = Ticks64;
+
+        fn now(&self) -> Self::Ticks {
+            0u32.into()
+        }
+    }
+
+    #[test]
+    fn test_from_ticks64() {
+        let s = Test1MHz64().ticks_to_seconds(1_000_000u32.into());
+        assert_eq!(s, 1);
+
+        let ms = Test1MHz64().ticks_to_ms(1_000_000u32.into());
+        assert_eq!(ms, 1_000);
+
+        let us = Test1MHz64().ticks_to_us(1_000_000u32.into());
+        assert_eq!(us, 1_000_000);
+
+        let s = Test1MHz64().ticks_to_seconds((1_000_000u64 << 31).into());
+        assert_eq!(s, 1 << 31);
+
+        let ms = Test1MHz64().ticks_to_ms((1_000_000u64 << 31).into());
+        assert_eq!(ms, !0u32);
+
+        let us = Test1MHz64().ticks_to_us((1_000_000u64 << 31).into());
+        assert_eq!(us, !0u32);
+    }
+
+    #[test]
+    fn test_to_ticks64() {
+        let t = Test1MHz64().ticks_from_seconds(1);
+        assert_eq!(t.into_u32(), 1_000_000);
+
+        let t = Test1MHz64().ticks_from_ms(1);
+        assert_eq!(t.into_u32(), 1_000);
+
+        let t = Test1MHz64().ticks_from_us(1);
+        assert_eq!(t.into_u32(), 1);
+
+        let t = Test1MHz64().ticks_from_seconds(1 << 31);
```

Comment:
It is also worth nothing that this test would have failed before this commit, the result would have been `u64::MAX` instead of `1_000_000u64 << 31`